transactions.md

```diff
@@ -3,4 +3,137 @@ title: Transactions
+   ```
+   If the application does not want to retry the transaction, for whatever
+   reason, it can simply `ROLLBACK` at this point. Any other statement will be
+   rejected by the server, as is generally the case after an error has been
+   encountered and the transaction has not been closed.
+5. When the transaction is ready to commit, use the standard `RELEASE SAVEPOINT` 
+   statement for committing the changes. If this succeeds, all changes made by the 
+   transaction become visible to others and are guaranteed to be durable if a crash occurs
+   (note that this is a departure from standard SQL semantics, where `RELEASE`
+   only destroyes a savepoint, but the changes can still be rolled back with the
+   transaction).
+   The `RELEASE SAVEPOINT` statement can itself fail with a retryable error. In
+   fact, in CockroachDB, `RELEASE` is the most likely statement to fail in this way, as,
+   under various circumstances, transactions only realize that they need to be restarted 
+   when they attempt to commit. If this happens, the error can be handled as
+   described in step 4.
+   If the `RELEASE` succeeds, the transaction can be finalized by issuing a `COMMIT`. This statement
+   doesn't have an effect over the data; it only serves to finalize a
+   transaction's state machine. `COMMIT` will not return a retryable error.
+   ```
+   RELEASE SAVEPOINT cockroach_restart
+   if retryable_error:
+      ROLLBACK TO SAVEPOINT cockroach_restart
+      ... execute the transaction again ...
+   COMMIT
+   ```
+
+Note that *implicit transactions* (SQL statements outside of a transaction) are internally 
+retried as needed; no special handling of errors is needed.
+
+## Transaction Priorities
+
+Each transaction in CockroachDB has a *priority*, assigned at start and possibly
+increased with each retry. This mechanism is designed to prevent starvation of
+certain classes of transaction under different access patterns. When two
+transactions interact with each other and one of them needs to be retried, the
+one with the lowest priority loses. But when it retries, it inherits the
+priority of the winner, meaning that each retry makes it stronger and more
+likely to succeed.  
+The client has some control over the initial priority of a transaction either
+when the transaction is started, using the `BEGIN TRANSACTION PRIORITY {LOW,
+NORMAL, HIGH}` statement, or immediately after using the `SET TRANSACTION PRIORITY {LOW,
+NORMAL, HIGH}` statement. These priorities can be assigned to transactions to
+help the ones with stricter deadlines succeed faster in high-contention
+scenarios.
+
+
+## Transaction Conflicts
+
+TODO
+
+## Isolation Levels
+
+CockroachDB internally supports two transaction isolation levels: `SNAPSHOT
+ISOLATION` and `SERIALIZABLE` (the default). These are mapped to the ANSI SQL
+isolation levels as follows: READ UNCOMMITTED, READ COMMITTED, SNAPSHOT ->
+SNAPSHOT ISOLATION.  REPEATABLE READ, SERIALIZABLE -> SERIALIZABLE. The
```

Comment:
Why wouldn't REPEATABLE READ map to SNAPSHOT ISOLATION? Seems the better choice. REPEATABLE READ has all of the same anomalies as SNAPSHOT plus some others (phantom reads).

And (you don't need to mention this here) Oracle's SERIALIZABLE actually maps to SNAPSHOT ISOLATION.